File: Go/Linked List/1721_Swapping Nodes in a Linked List.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func swapNodes(head *ListNode, k int) *ListNode {
	cur := head
	count := 1
	for count < k {
		count += 1
		cur = cur.Next
	}
	temp := head
	for next := cur.Next; next != nil; next = next.Next {
		temp = temp.Next
	}
	cur.Val, temp.Val = temp.Val, cur.Val
	return head
}

File: C++/Linked List/1721_Swapping Nodes_in_a_LinkedList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution
{
public:
    ListNode *swapNodes(ListNode *head, int k)
    {
        ListNode *dummy = new ListNode(0, head);
        ListNode *first = dummy, *sec = dummy;
        ListNode *temp;
        for (int i = 1; i <= k + 1; i++)
        {
            sec = sec->next;
            if (i == k)
            {
                temp = sec;
            }
        }
        while (sec != NULL)
        {
            sec = sec->next;
            first = first->next;
        }
        int tempVal = first->next->val;
        first->next->val = temp->val;
        temp->val = tempVal;
        return head;
    }
};
